tests/accessor/local_accessor_api_common.h

```diff
@@ -0,0 +1,208 @@
+  res_acc[7, item_id] =
+      value_helper::are_equal(*acc_multi_ptr_yes.get(), expected_data);
+
+  auto acc_pointer = accessor.get_pointer();
+  res_acc[8, item_id] =
+      std::is_same_v<decltype(acc_pointer),
+                     std::add_pointer_t<typename AccT::value_type>>;
+  res_acc[9, item_id] = value_helper::are_equal(*acc_pointer, expected_data);
+}
+
+template <typename T, typename DimensionTypeT>
+class run_api_tests {
+  static constexpr int dims = DimensionTypeT::value;
+  using AccT = sycl::local_accessor<T, dims>;
+
+ public:
+  void operator()(const std::string &type_name) {
+    auto queue = util::get_cts_object::queue();
+
+    SECTION(
+        get_section_name<dims>(type_name, "Check local_accessor alias types")) {
+      test_local_accessor_types<T, AccT, AccessModeT, TargetT>();
+    }
+
+    SECTION(get_section_name<dims>(type_name,
+                                   "Check api for empty local_accessor")) {
+      queue
+          .submit([&](sycl::handler &cgh) {
+            AccT acc;
+            test_accessor_methods_common(acc, 0 /* expected_byte_size*/,
+                                         0 /*expected_size*/,
+                                         util::get_cts_object::range<dims>::get(
+                                             0, 0, 0) /*expected_range*/);
+          })
+          .wait_and_throw();
+    }
+
+    SECTION(get_section_name<dims>(type_name, access_mode_name, target_name,
+                                   "Check api for local_accessor")) {
+      constexpr size_t global_range_size = 4;
+      constexpr size_t local_range_size = 2;
+      auto global_range = util::get_cts_object::range<dims>::get(
+          global_range_size, global_range_size, global_range_size);
+      auto local_range = util::get_cts_object::range<dims>::get(
+          local_range_size, local_range_size, local_range_size);
+      sycl::nd_range<dim> nd_range(global_range, local_range);
+      constexpr size_t checks_count = 11;
+      bool res[checks_count, global_range.size()];
+      std::fill(res, (res + checks_count * global_range.size()), true);
+      {
+        sycl::buffer res_buf(res,
+                             sycl::range<2>(checks_count, global_range.size()));
+        queue
+            .submit([&](sycl::handler &cgh) {
+              AccT acc(local_range, cgh);
+              AccT acc_other(local_range, cgh);
+
+              test_accessor_methods_common(
+                  acc, sizeof(T) * local_range.size() /* expected_byte_size*/,
+                  local_range.size() /*expected_size*/,
+                  local_range /*expected_range*/);
+
+              sycl::accessor res_acc(res_buf, cgh);
+              cgh.parallel_for(nd_range, [=](sycl::nd_item<dim> item) {
+                auto ref_1 = acc[0];
+
+                auto id = util::get_cts_object::id<dims>::get(1, 1, 1);
+                auto ref_2 = get_subscript_overload<T, AccT, dims>(acc, id);
```

Comment:
```suggestion
                auto&& ref_2 = get_subscript_overload<T, AccT, dims>(acc, id);
```
Is it what you've meant? Note that `auto` would resolve to the non-const non-reference variable no matter of `get_subscript_overload` return type.